bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBackyardHandler.java

```diff
@@ -0,0 +1,117 @@
+
+    public HaywardBackyardHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void getTelemetry(String xmlResponse) throws Exception {
+        List<String> data = new ArrayList<>();
+        List<String> systemIDs = new ArrayList<>();
+
+        @SuppressWarnings("null")
+        HaywardBridgeHandler bridgehandler = (HaywardBridgeHandler) getBridge().getHandler();
+        if (bridgehandler != null) {
+            systemIDs = bridgehandler.evaluateXPath("//Backyard/@systemId", xmlResponse);
+            String thingSystemID = getThing().getUID().getId();
+            for (int i = 0; i < systemIDs.size(); i++) {
+                if (systemIDs.get(i).equals(thingSystemID)) {
+                    // Air temp
+                    data = bridgehandler.evaluateXPath("//Backyard/@airTemp", xmlResponse);
+                    updateData(HaywardBindingConstants.CHANNEL_BACKYARD_AIRTEMP, data.get(0));
+
+                    // Status
+                    data = bridgehandler.evaluateXPath("//Backyard/@status", xmlResponse);
+                    updateData(HaywardBindingConstants.CHANNEL_BACKYARD_STATUS, data.get(0));
+
+                    // State
+                    data = bridgehandler.evaluateXPath("//Backyard/@state", xmlResponse);
+                    updateData(HaywardBindingConstants.CHANNEL_BACKYARD_STATE, data.get(0));
+                }
+            }
+        }
+    }
+
+    public boolean getAlarmList(String systemID) throws Exception {
+        List<String> bowID = new ArrayList<>();
+        List<String> parameter1 = new ArrayList<>();
+        List<String> message = new ArrayList<>();
+        String alarmStr;
+
+        @SuppressWarnings("null")
+        HaywardBridgeHandler bridgehandler = (HaywardBridgeHandler) getBridge().getHandler();
+
+        // *****Request Alarm List from Hayward server
+        @SuppressWarnings("null")
+        String urlParameters = "<?xml version=\"1.0\" encoding=\"utf-8\"?><Request><Name>GetAlarmList</Name><Parameters>"
+                + "<Parameter name=\"Token\" dataType=\"String\">" + bridgehandler.account.token + "</Parameter>"
+                + "<Parameter name=\"MspSystemID\" dataType=\"int\">" + bridgehandler.account.mspSystemID
+                + "</Parameter>"
+                + "<Parameter name=\"CultureInfoName\" dataType=\"String\">en-us</Parameter></Parameters></Request>";
+
+        String xmlResponse = bridgehandler.httpXmlResponse(urlParameters);
+
+        if (xmlResponse.isEmpty()) {
+            logger.debug("Hayward getAlarmList XML response was empty");
+            return false;
+        }
+
+        String status = bridgehandler
+                .evaluateXPath("/Response/Parameters//Parameter[@name='Status']/text()", xmlResponse).get(0);
+
+        if (!(status.equals("0"))) {
+            logger.trace("Hayward getAlarm XML response: {}", xmlResponse);
+            return false;
+        }
+
+        if (status.equals("0")) {
```

Comment:
This if statement is redundant since the code above would return if `!(status.equals("0")`